contracts/wallet/BaseWallet.sol

```diff
@@ -27,7 +27,7 @@ import "./IWallet.sol";
 contract BaseWallet is IWallet {
 
     // The implementation of the proxy
-    address public implementation;
+    address public override implementation;
```

Comment:
Note that this slightly changes the bytecode of the BaseWallet, so maybe it'd be better to keep BaseWallet unchanged and use `BaseWallet(_wallet).implementation()` in `ArgentWalletDetector.sol` instead of `IWallet(_wallet).implementation()`.

Reply:
Are you sure it changes the bytecode? My understanding is that `override` is just used by the compiler to enforce a specific behaviour.

I know I could use BaseWallet instead but would prefer to code against the interface.

Reply:
Yes, I checked it both in Remix and in etherlime. It changes 32 bytes close to the end of the bytecode.

Reply:
I really think we should use an interface here and not a specific `BaseWallet` implementation. Should I define a new IWallet interface locally until the next time we update the BaseWallet?

@elenadimitrova any opinion on this?

Reply:
A local interface definition would work, though I think you'd need to call is something different than IWallet to avoid conflicts in the build directory